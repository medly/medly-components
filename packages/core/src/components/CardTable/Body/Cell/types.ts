import { ColumnConfig, Data } from '../../types';

export interface Props {
    data: any;
    column: ColumnConfig;
    rowId: any;
    isRowClickDisabled?: boolean;
    rowData: Data;
}

export interface StyledProps {
    align?: 'left' | 'center' | 'right';
    wrapText?: boolean;
}
