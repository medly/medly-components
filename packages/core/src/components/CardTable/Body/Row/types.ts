import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    onClick?: () => void;
    isRowClickDisabled?: boolean;
    withWhiteBackground?: boolean;
    gridTemplateColumns: string;
}
