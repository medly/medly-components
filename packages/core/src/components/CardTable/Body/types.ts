import { ColumnConfig, Data } from '../types';

export interface Props {
    data: Data[];
    columns: ColumnConfig[];
    onRowClick?: (rowData: Record<string, unknown>) => void;
    rowClickDisableKey?: string;
    uniqueKeyName?: string;
    withWhiteBackground?: boolean;
}
