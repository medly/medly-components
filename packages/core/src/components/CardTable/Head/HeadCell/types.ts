import { ColumnConfig, SortOrder } from '../../types';

export interface Props {
    defaultSortOrder?: SortOrder | 'none';
    column: ColumnConfig;
    sortField?: string;
    withWhiteBackground?: boolean;
    onSortChange?: (field: string, order: SortOrder) => void;
}

export interface StyledProps {
    isActive?: boolean;
    sortable?: boolean;
    withWhiteBackground?: boolean;
    align?: 'left' | 'center' | 'right';
}
