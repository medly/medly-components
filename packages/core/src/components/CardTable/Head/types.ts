import { ColumnConfig, SortOrder } from '../types';

export interface Props {
    defaultSortField?: string;
    defaultSortOrder?: SortOrder;
    withWhiteBackground?: boolean;
    columns: ColumnConfig[];
    onSort?: (field: string, order: SortOrder) => void;
}
