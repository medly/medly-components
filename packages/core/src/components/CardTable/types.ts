import { HTMLProps } from '@medly-components/utils';
import type { FC } from 'react';

export type SortOrder = 'asc' | 'desc';

export interface Data {
    [key: string]: any;
}

export interface ColumnConfig {
    /** Title of the column */
    title: string;
    /** Field name in the data */
    field: string;
    /** Set it true to show sort icons */
    sortable?: boolean;
    /** Column content alignment */
    align?: 'left' | 'right' | 'center';
    /** Column Text content wrapping */
    wrapText?: boolean;
    /** Provide the fraction of the of the total width of the row */
    fraction?: number;
    /** Custom component */
    component?: FC<{ data: any; rowId?: any; disabled?: boolean; rowData?: Data }>;
}
export interface CardTableProps extends Omit<HTMLProps<HTMLOListElement>, 'data' | 'type'> {
    /** Table data */
    data: Array<{
        [key: string]: any;
    }>;
    /** Column configuration */
    columns: ColumnConfig[];
    /** Key name in data to be used as unique id */
    uniqueKeyName?: string;
    /** Default Sort Field*/
    defaultSortField?: string;
    /** Default Sort Order*/
    defaultSortOrder?: SortOrder;
    /** Function to be called on click of sort icon */
    onSort?: (field: string, order: SortOrder) => void;
    /** Function to be called on row click */
    onRowClick?: (rowData: Record<string, unknown>) => void;
    /** Key name to disable row click */
    rowClickDisableKey?: string;
    /** Key name to select background color */
    withWhiteBackground?: boolean;
}
