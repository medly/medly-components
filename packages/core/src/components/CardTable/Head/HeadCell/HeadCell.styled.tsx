import { SvgIcon } from '@medly-components/icons';
import styled from 'styled-components';
import Text from '../../../Text';
import { StyledProps } from './types';

export const HeadCellContent = styled('span')`
    padding: 1rem 1.6rem;
    display: flex;
    align-items: center;
`;

export const HeadCell = styled('th').attrs(({ withWhiteBackground }: StyledProps) => ({
    bgTheme: withWhiteBackground ? 'lightBackground' : 'darkBackground'
}))<StyledProps & { bgTheme: 'lightBackground' | 'darkBackground' }>`
    padding: 0 0.8rem;
    margin: 1.4rem 0;
    display: flex;
    align-items: center;
    position: relative;
    justify-content: ${({ align }) => (align === 'left' ? 'flex-start' : align === 'right' ? 'flex-end' : 'center')};

    &::after {
        content: '';
        width: calc(100% - 1.6rem);
        margin: 0 0.8rem;
        display: block;
        position: absolute;
        bottom: -1.6rem;
        left: 0;
        background-color: ${({ theme, bgTheme, isActive }) => (isActive ? theme.cardTable[bgTheme].headerActiveColor : 'transparent')};
        height: 0.2rem;
        transition: all 100ms ease-out;
        border-radius: 0.2rem 0.2rem 0 0;
    }

    ${HeadCellContent} {
        &:hover {
            cursor: ${({ sortable }) => (sortable ? 'pointer' : 'default')};
            ${Text.Style} {
                color: ${({ theme, bgTheme }) => theme.cardTable[bgTheme].headerHoverColor};
            }
            ${SvgIcon} {
                * {
                    fill: ${({ theme, bgTheme }) => theme.cardTable[bgTheme].headerHoverColor};
                }
            }
        }
    }

    ${Text.Style} {
        transition: all 100ms ease-out;
        user-select: none;
        display: inline;
        color: ${({ theme, bgTheme, isActive }) =>
            isActive ? theme.cardTable[bgTheme].headerActiveColor : theme.cardTable[bgTheme].headerColor};
    }

    ${SvgIcon} {
        transition: all 100ms ease-out;
        width: 2.4rem;
        cursor: pointer;
        padding-left: 0.4rem;
        * {
            fill: ${({ theme, bgTheme, isActive }) =>
                isActive ? theme.cardTable[bgTheme].headerActiveColor : theme.cardTable[bgTheme].headerColor};
        }
    }
`;
HeadCell.defaultProps = { align: 'left' };
