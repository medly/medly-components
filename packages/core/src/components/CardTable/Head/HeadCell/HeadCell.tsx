import { ChevronDownIcon, ChevronUpIcon, DropdownIcon } from '@medly-components/icons';
import { WithStyle } from '@medly-components/utils';
import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
import Text from '../../../Text';
import * as Styled from './HeadCell.styled';
import { Props } from './types';

export const Component: FC<Props> = React.memo(({ defaultSortOrder, sortField, onSortChange, withWhiteBackground, column }) => {
    const { field, sortable, title, align } = column,
        [sortState, setSortState] = useState<'none' | 'asc' | 'desc'>(defaultSortOrder!);

    useEffect(() => {
        sortField !== field && setSortState('none');
    }, [sortField]);

    const handleSort = useCallback(() => {
        const order = sortState === 'asc' ? 'desc' : 'asc';
        setSortState(order);
        onSortChange && onSortChange(field, order);
    }, [sortState, onSortChange, field]);

    const sortIcon = useMemo(
        () =>
            sortField !== field ? (
                <DropdownIcon size="S" />
            ) : sortState === 'desc' ? (
                <ChevronDownIcon size="M" />
            ) : (
                <ChevronUpIcon size="M" />
            ),
        [handleSort, sortField, field, sortState]
    );

    return (
        <Styled.HeadCell isActive={sortField === field} {...{ align, sortable, withWhiteBackground }}>
            <Styled.HeadCellContent onClick={sortable ? handleSort : undefined}>
                <Text uppercase textVariant="h5">
                    {title}
                </Text>
                {sortable && sortIcon}
            </Styled.HeadCellContent>
        </Styled.HeadCell>
    );
});
Component.defaultProps = {
    defaultSortOrder: 'none'
};
Component.displayName = 'HeadCell';

export const HeadCell: React.FC<Props> & WithStyle = Object.assign(Component, { Style: Styled.HeadCell });
