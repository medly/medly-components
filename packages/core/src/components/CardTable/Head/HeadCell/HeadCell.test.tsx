import { fireEvent, render } from '@test-utils';
import React from 'react';
import { ColumnConfig } from '../../types';
import { HeadCell } from './HeadCell';
import { StyledProps } from './types';

const dummyColumn: ColumnConfig = { title: 'Demo', field: 'demo', sortable: true },
    renderer = ({ withWhiteBackground = false, sortField = 'random', onSortChange = jest.fn(), column = dummyColumn }) =>
        render(
            <table>
                <thead>
                    <tr>
                        <HeadCell
                            sortField={sortField}
                            onSortChange={onSortChange}
                            withWhiteBackground={withWhiteBackground}
                            column={column}
                        />
                    </tr>
                </thead>
            </table>
        );

describe('Card Table HeadCell', () => {
    it('should render properly with dark background and dropdown icon by default', () => {
        const { container } = renderer({ withWhiteBackground: false, column: { title: 'Demo', field: 'demo' } });
        expect(container).toMatchSnapshot();
    });

    it('should render properly with light background and dropdown icon by default', () => {
        const { container } = renderer({ withWhiteBackground: true });
        expect(container).toMatchSnapshot();
    });

    it('should render properly with hover state', () => {
        const { container, getByText } = renderer({ withWhiteBackground: true });
        fireEvent.mouseEnter(getByText('Demo'));
        expect(container).toMatchSnapshot();
    });

    it('should call onSortChange with asc order when clicked on sorting icon', () => {
        const mockOnSortChange = jest.fn(),
            { container } = renderer({ onSortChange: mockOnSortChange, sortField: 'demo' });

        fireEvent.click(container.querySelectorAll('svg')[0]);
        expect(container).toMatchSnapshot();
        expect(mockOnSortChange).toHaveBeenCalledWith('demo', 'asc');
    });

    it('should call onSortChange with desc order when clicked on sorting icon', () => {
        const mockOnSortChange = jest.fn(),
            { container } = renderer({ onSortChange: mockOnSortChange, sortField: 'demo' });

        fireEvent.click(container.querySelectorAll('svg')[0]);
        fireEvent.click(container.querySelectorAll('svg')[0]);
        expect(container).toMatchSnapshot();
        expect(mockOnSortChange).toHaveBeenCalledWith('demo', 'desc');
    });

    const positions: Required<StyledProps>['align'][] = ['left', 'center', 'right'];

    test.each(positions)('should be able to %p align ', align => {
        const column: ColumnConfig = {
                align,
                title: 'Column 1',
                field: 'column1'
            },
            { container } = renderer({ column });
        expect(container).toMatchSnapshot();
    });
});
