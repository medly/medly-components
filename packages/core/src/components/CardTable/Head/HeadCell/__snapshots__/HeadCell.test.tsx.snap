// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Table HeadCell should be able to "center" align 1`] = `
.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: initial;
  text-transform: uppercase;
}

.c2 {
  padding: 1rem 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  padding: 0 0.8rem;
  margin: 1.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0::after {
  content: '';
  width: calc(100% - 1.6rem);
  margin: 0 0.8rem;
  display: block;
  position: absolute;
  bottom: -1.6rem;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1:hover {
  cursor: default;
}

.c0 .c1:hover .c3 {
  color: #5687D9;
}

.c0 .c1:hover .sc-1pf3x8a-0 * {
  fill: #5687D9;
}

.c0 .c3 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline;
  color: #435465;
}

.c0 .sc-1pf3x8a-0 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  width: 2.4rem;
  cursor: pointer;
  padding-left: 0.4rem;
}

.c0 .sc-1pf3x8a-0 * {
  fill: #435465;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <span
            class="c1 c2"
          >
            <h5
              class="c3 c4"
            >
              Column 1
            </h5>
          </span>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should be able to "left" align 1`] = `
.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: initial;
  text-transform: uppercase;
}

.c2 {
  padding: 1rem 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  padding: 0 0.8rem;
  margin: 1.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: calc(100% - 1.6rem);
  margin: 0 0.8rem;
  display: block;
  position: absolute;
  bottom: -1.6rem;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1:hover {
  cursor: default;
}

.c0 .c1:hover .c3 {
  color: #5687D9;
}

.c0 .c1:hover .sc-1pf3x8a-0 * {
  fill: #5687D9;
}

.c0 .c3 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline;
  color: #435465;
}

.c0 .sc-1pf3x8a-0 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  width: 2.4rem;
  cursor: pointer;
  padding-left: 0.4rem;
}

.c0 .sc-1pf3x8a-0 * {
  fill: #435465;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <span
            class="c1 c2"
          >
            <h5
              class="c3 c4"
            >
              Column 1
            </h5>
          </span>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should be able to "right" align 1`] = `
.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: initial;
  text-transform: uppercase;
}

.c2 {
  padding: 1rem 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  padding: 0 0.8rem;
  margin: 1.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0::after {
  content: '';
  width: calc(100% - 1.6rem);
  margin: 0 0.8rem;
  display: block;
  position: absolute;
  bottom: -1.6rem;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1:hover {
  cursor: default;
}

.c0 .c1:hover .c3 {
  color: #5687D9;
}

.c0 .c1:hover .sc-1pf3x8a-0 * {
  fill: #5687D9;
}

.c0 .c3 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline;
  color: #435465;
}

.c0 .sc-1pf3x8a-0 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  width: 2.4rem;
  cursor: pointer;
  padding-left: 0.4rem;
}

.c0 .sc-1pf3x8a-0 * {
  fill: #435465;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <span
            class="c1 c2"
          >
            <h5
              class="c3 c4"
            >
              Column 1
            </h5>
          </span>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should call onSortChange with asc order when clicked on sorting icon 1`] = `
.c6 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c6 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: initial;
  text-transform: uppercase;
}

.c2 {
  padding: 1rem 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  padding: 0 0.8rem;
  margin: 1.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: calc(100% - 1.6rem);
  margin: 0 0.8rem;
  display: block;
  position: absolute;
  bottom: -1.6rem;
  left: 0;
  background-color: #3872D2;
  height: 0.2rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1:hover {
  cursor: pointer;
}

.c0 .c1:hover .c3 {
  color: #5687D9;
}

.c0 .c1:hover .c5 * {
  fill: #5687D9;
}

.c0 .c3 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline;
  color: #3872D2;
}

.c0 .c5 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  width: 2.4rem;
  cursor: pointer;
  padding-left: 0.4rem;
}

.c0 .c5 * {
  fill: #3872D2;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <span
            class="c1 c2"
          >
            <h5
              class="c3 c4"
            >
              Demo
            </h5>
            <svg
              class="c5 c6"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.208 9.03a1 1 0 0 1 1.584 0l3.165 4.11a1 1 0 0 1-.793 1.61H8.836a1 1 0 0 1-.793-1.61l3.165-4.11z"
                fill="#000"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should call onSortChange with desc order when clicked on sorting icon 1`] = `
.c6 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c6 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: initial;
  text-transform: uppercase;
}

.c2 {
  padding: 1rem 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  padding: 0 0.8rem;
  margin: 1.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: calc(100% - 1.6rem);
  margin: 0 0.8rem;
  display: block;
  position: absolute;
  bottom: -1.6rem;
  left: 0;
  background-color: #3872D2;
  height: 0.2rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1:hover {
  cursor: pointer;
}

.c0 .c1:hover .c3 {
  color: #5687D9;
}

.c0 .c1:hover .c5 * {
  fill: #5687D9;
}

.c0 .c3 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline;
  color: #3872D2;
}

.c0 .c5 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  width: 2.4rem;
  cursor: pointer;
  padding-left: 0.4rem;
}

.c0 .c5 * {
  fill: #3872D2;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <span
            class="c1 c2"
          >
            <h5
              class="c3 c4"
            >
              Demo
            </h5>
            <svg
              class="c5 c6"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                fill="#000"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should render properly with dark background and dropdown icon by default 1`] = `
.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: initial;
  text-transform: uppercase;
}

.c2 {
  padding: 1rem 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  padding: 0 0.8rem;
  margin: 1.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: calc(100% - 1.6rem);
  margin: 0 0.8rem;
  display: block;
  position: absolute;
  bottom: -1.6rem;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1:hover {
  cursor: default;
}

.c0 .c1:hover .c3 {
  color: #5687D9;
}

.c0 .c1:hover .sc-1pf3x8a-0 * {
  fill: #5687D9;
}

.c0 .c3 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline;
  color: #435465;
}

.c0 .sc-1pf3x8a-0 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  width: 2.4rem;
  cursor: pointer;
  padding-left: 0.4rem;
}

.c0 .sc-1pf3x8a-0 * {
  fill: #435465;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <span
            class="c1 c2"
          >
            <h5
              class="c3 c4"
            >
              Demo
            </h5>
          </span>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should render properly with hover state 1`] = `
.c6 {
  overflow: visible;
  font-size: 2rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c6 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: initial;
  text-transform: uppercase;
}

.c2 {
  padding: 1rem 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  padding: 0 0.8rem;
  margin: 1.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: calc(100% - 1.6rem);
  margin: 0 0.8rem;
  display: block;
  position: absolute;
  bottom: -1.6rem;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1:hover {
  cursor: pointer;
}

.c0 .c1:hover .c3 {
  color: #5687D9;
}

.c0 .c1:hover .c5 * {
  fill: #5687D9;
}

.c0 .c3 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline;
  color: #607890;
}

.c0 .c5 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  width: 2.4rem;
  cursor: pointer;
  padding-left: 0.4rem;
}

.c0 .c5 * {
  fill: #607890;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <span
            class="c1 c2"
          >
            <h5
              class="c3 c4"
            >
              Demo
            </h5>
            <svg
              class="c5 c6"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.229 6.935a1 1 0 0 1 1.543 0l2.208 2.679a1 1 0 0 1-.77 1.636H9.79a1 1 0 0 1-.77-1.636l2.209-2.679zM12.771 17.065a1 1 0 0 1-1.543 0L9.02 14.386a1 1 0 0 1 .772-1.636h4.418a1 1 0 0 1 .771 1.636l-2.209 2.679z"
                fill="#435465"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should render properly with light background and dropdown icon by default 1`] = `
.c6 {
  overflow: visible;
  font-size: 2rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c6 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: initial;
  text-transform: uppercase;
}

.c2 {
  padding: 1rem 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  padding: 0 0.8rem;
  margin: 1.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: calc(100% - 1.6rem);
  margin: 0 0.8rem;
  display: block;
  position: absolute;
  bottom: -1.6rem;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1:hover {
  cursor: pointer;
}

.c0 .c1:hover .c3 {
  color: #5687D9;
}

.c0 .c1:hover .c5 * {
  fill: #5687D9;
}

.c0 .c3 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline;
  color: #607890;
}

.c0 .c5 {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  width: 2.4rem;
  cursor: pointer;
  padding-left: 0.4rem;
}

.c0 .c5 * {
  fill: #607890;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <span
            class="c1 c2"
          >
            <h5
              class="c3 c4"
            >
              Demo
            </h5>
            <svg
              class="c5 c6"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.229 6.935a1 1 0 0 1 1.543 0l2.208 2.679a1 1 0 0 1-.77 1.636H9.79a1 1 0 0 1-.77-1.636l2.209-2.679zM12.771 17.065a1 1 0 0 1-1.543 0L9.02 14.386a1 1 0 0 1 .772-1.636h4.418a1 1 0 0 1 .771 1.636l-2.209 2.679z"
                fill="#435465"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
